tests/test-cases/Group1-Docker-Commands/1-07-Docker-Stop.robot

```diff
@@ -50,12 +50,18 @@ Assert Kill Signal
     Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Should Be Equal  ${output}  poweredOff
     ${rc}  ${dir}=  Run Keyword If  '%{DATASTORE_TYPE}' == 'VSAN'  Run And Return Rc And Output  govc datastore.ls ${vmName}*
     Run Keyword If  '%{DATASTORE_TYPE}' == 'VSAN'  Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${output}=  Run Keyword If  '%{DATASTORE_TYPE}' == 'VSAN'  Run And Return Rc And Output  govc datastore.download ${dir}/tether.debug -
+    Run Keyword If  '%{DATASTORE_TYPE}' == 'VSAN'  Should Be Equal As Integers  ${rc}  0
+    Run Keyword If  '%{DATASTORE_TYPE}' == 'VSAN'  Run Keyword If  ${expect}  Should Contain  ${output}  sending signal KILL
+    Run Keyword If  '%{DATASTORE_TYPE}' == 'VSAN'  Run Keyword Unless  ${expect}  Should Not Contain  ${output}  sending signal KILL
+
     ${rc}  ${dir}=  Run Keyword If  '%{DATASTORE_TYPE}' == 'Non_VSAN'  Run And Return Rc And Output  govc datastore.ls ${id}*
     Run Keyword If  '%{DATASTORE_TYPE}' == 'Non_VSAN'  Should Be Equal As Integers  ${rc}  0
-    ${rc}  ${output}=  Run And Return Rc And Output  govc datastore.download ${dir}/tether.debug -
-    Should Be Equal As Integers  ${rc}  0
-    Run Keyword If  ${expect}  Should Contain  ${output}  sending signal KILL
-    Run Keyword Unless  ${expect}  Should Not Contain  ${output}  sending signal KILL
+
+    ${rc}  ${output}=  Run Keyword If  '%{DATASTORE_TYPE}' == 'Non_VSAN'  Run And Return Rc And Output  govc datastore.download ${dir}/tether.debug -
+    Run Keyword If  '%{DATASTORE_TYPE}' == 'Non_VSAN'  Should Be Equal As Integers  ${rc}  0
+    Run Keyword If  '%{DATASTORE_TYPE}' == 'Non_VSAN'  Run Keyword If  ${expect}  Should Contain  ${output}  sending signal KILL
+    Run Keyword If  '%{DATASTORE_TYPE}' == 'Non_VSAN'  Run Keyword Unless  ${expect}  Should Not Contain  ${output}  sending signal KILL
```

Comment:
This all is getting a bit clunky at this point, AFAICT the only difference is the vm name in this procedure? Can you break this out to a custom keyword instead that just takes the vm name/id and does this so we can make this a bit more readable?